### ISSUE TITLE
require-atomic-updates false report

### ISSUE THREAD

**Issue by aladdin-add** [eslint/eslint#11950]
**Tell us about your environment**

* **ESLint Version:** 6.0.1
* **Node Version:** v12
* **npm Version:**

**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
rules: {require-atomic-updates: 2}
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
async function tryFetchMainfest() {
  const config = await pureFetchMainfest(flasher.manifestId);
  if (config) {
    flasher.config = config;
  }
}
```

```bash

```

**What did you expect to happen?**
no error reported.

**What actually happened? Please include the actual, raw output from ESLint.**
 45:5   error    Possible race condition: `flasher.config` might be reassigned based on an outdated value of `flasher.config`  require-atomic-updates

**Are you willing to submit a pull request to fix this bug?**
not sure.

**Comment by mysticatea** [comment]
I think this is a duplicate of #11899.
